WooCommerce/Classes/ServiceLocator/Analytics.swift

```diff
@@ -29,6 +29,15 @@ protocol Analytics {
     ///
     func track(_ stat: WooAnalyticsStat, withError error: Error)
 
+    /// Track a specific event with associated properties and an associated error (that is translated to properties)
+    ///
+    /// - Parameters:
+    ///   - stat: the event name
+    ///   - properties: a collection of properties related to the event
+    ///   - error: the error to track
+    ///
+    func track(_ stat: WooAnalyticsStat, properties: [AnyHashable: Any]?, withError error: Error?)
```

Comment:
nit: just a suggestion
```suggestion
    func track(_ stat: WooAnalyticsStat, properties: [AnyHashable: Any]?, error: Error?)
```

Reply:
Yeah, agreed. I'd included `withError` to be consistent with the existing methods but just because the ones that are already there are too wordy, doesn't mean these need to be.